File: 1.1/src/cz/mg/vulkan/PFNvkFreeDescriptorSets.java
package cz.mg.vulkan;

public class PFNvkFreeDescriptorSets extends VkFunctionPointer {
    public PFNvkFreeDescriptorSets() {
    }

    protected PFNvkFreeDescriptorSets(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkFreeDescriptorSets(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkFreeDescriptorSets(long value) {
        setValue(value);
    }

    public PFNvkFreeDescriptorSets(VkInstance instance) {
        super(instance, new VkString("vkFreeDescriptorSets"));
    }

    public void call(VkDevice device, VkDescriptorPool descriptorPool, VkUInt32 descriptorSetCount, VkDescriptorSet pDescriptorSets, VkResult rval){
        callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), descriptorPool != null ? descriptorPool.getVkAddress() : VkPointer.getNullAddressNative(), descriptorSetCount != null ? descriptorSetCount.getVkAddress() : VkPointer.getNullAddressNative(), pDescriptorSets != null ? pDescriptorSets.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getSinkAddressNative());
    }

    protected static native void callNative(long vkaddress, long device, long descriptorPool, long descriptorSetCount, long pDescriptorSets, long rval);


    public int call(VkDevice device, VkDescriptorPool descriptorPool, int descriptorSetCount, VkDescriptorSet pDescriptorSets){
        return callSimplifiedNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), descriptorPool != null ? descriptorPool.getVkAddress() : VkPointer.getNullAddressNative(), descriptorSetCount, pDescriptorSets != null ? pDescriptorSets.getVkAddress() : VkPointer.NULL);
    }

    protected static native int callSimplifiedNative(long vkaddress, long device, long descriptorPool, int descriptorSetCount, long pDescriptorSets);

}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkFreeDescriptorSets.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkFreeDescriptorSets_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong descriptorPool, jlong descriptorSetCount, jlong pDescriptorSets, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkFreeDescriptorSets f = (PFN_vkFreeDescriptorSets)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkDescriptorPool*)jniLongToPointer(descriptorPool)),
        *((uint32_t*)jniLongToPointer(descriptorSetCount)),
        ((VkDescriptorSet*)jniLongToPointer(pDescriptorSets))
    );
}


jint Java_cz_mg_vulkan_PFNvkFreeDescriptorSets_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong descriptorPool, jint descriptorSetCount, jlong pDescriptorSets){
    (void)env;
    (void)jc;
    PFN_vkFreeDescriptorSets f = (PFN_vkFreeDescriptorSets)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkDescriptorPool*)jniLongToPointer(descriptorPool)),
        (uint32_t)descriptorSetCount,
        ((VkDescriptorSet*)jniLongToPointer(pDescriptorSets))
    );
}

